File: app/src/main/java/com/rowland/delivery/services/auth/models/Group.java
package com.rowland.delivery.services.auth.models;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by Rowland on 5/2/2018.
 */

public class Group {

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({MERCHANT, CLIENT})
    @interface GroupName {

    }

    public static final String MERCHANT = "merchant";
    public static final String CLIENT = "client";

    @GroupName
    public String name;


}

File: ui/src/main/java/com/rowland/delivery/merchant/features/auth/models/Group.kt
package com.rowland.delivery.merchant.features.auth.models

import androidx.annotation.StringDef
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by Rowland on 5/2/2018.
 */

class Group {

    @GroupName
    var name: String? = null

    @Retention(RetentionPolicy.SOURCE)
    @StringDef(MERCHANT, CLIENT)
    internal annotation class GroupName

    companion object {

        const val MERCHANT = "merchant"
        const val CLIENT = "client"
    }


}
